tools/omnissm/functions/subscribe/main.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2018 Capital One Services, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+
+	"github.com/capitalone/cloud-custodian/tools/omnissm/pkg/awsutil"
+	"github.com/capitalone/cloud-custodian/tools/omnissm/pkg/identity"
+	"github.com/capitalone/cloud-custodian/tools/omnissm/pkg/manager"
+)
+
+var (
+	resourceTypes = map[string]struct{}{
+		"AWS::EC2::Instance": struct{}{},
+	}
+
+	resourceStatusTypes = map[string]struct{}{
+		"ResourceDeleted":    struct{}{},
+		"ResourceDiscovered": struct{}{},
+		"OK":                 struct{}{},
+	}
+	resourceTags = make(map[string]struct{})
+
+	RegistrationsTable = os.Getenv("OMNISSM_REGISTRATIONS_TABLE")
+	ResourceTags       = os.Getenv("OMNISSM_RESOURCE_TAGS")
+)
+
+func init() {
+	if RegistrationsTable == "" {
+		RegistrationsTable = "omnissm-registrations"
+	}
+	if ResourceTags == "" {
+		ResourceTags = "App,OwnerContact,Name"
+	}
+}
+
+func handleConfigurationItemChange(detail manager.ConfigurationItemDetail) error {
+	cfg, err := awsutil.LoadDefaultAWSConfig()
```

Comment:
I'm not sure what steps the aws sdk takes to initialize a repeat client, but it could be significant for Go standards. I think it would be worth moving at least the config, if not the client as well into a global or attach it to a struct that this function becomes a method of. From my understanding the majority of the work done is in building the config/session which will actually have the most resources associated with it.

Reply:
Looking at this again I think the `manager.Manager` should just be created in `init()`.